keps/sig-node/1287-in-place-update-pod-resources/README.md

```diff
@@ -898,7 +899,48 @@ Pod.
 A previous version of kubelet interprets mutation to Pod Resources as a Container definition
 change and will restart the container with the new Resources. This could lead to Node resource
 over-subscription. In order to address this, the feature-gate will remain default false for
-atleast two versions after the initial release that carries it.
+atleast two versions after v1.27 alpha release. i.e: beta is planned for v1.29 and
+InPlacePodVeritcalScaling feature-gate will be true in versions v1.29+
+
+kubelet: When feature-gate is disabled, if kubelet sees a Proposed resize, it rejects the
+resize as Infeasible.
+
+scheduler: If PodStatus.Resize field is not empty, scheduler uses max(ResourcesAllocated, Requests)
+even if the feature-gate is disabled.
+
+Allowed [version skews](https://kubernetes.io/releases/version-skew-policy/) are handled as below:
+
+| apiserver ver -> |    v1.27    |    v1.28     |    v1.29    |    v1.30    |
+|------------------|-------------|--------------|-------------|-------------|
+| kubelet v1.25    |     N       |     X        |     X       |     X       |
+| kubelet v1.26    |     N       |     N        |     X       |     X       |
+| kubelet v1.27    |     N       |     N        |     A       |     X       |
+| kubelet v1.28    |     X       |     N        |     A       |     A       |
+| kubelet v1.29    |     X       |     X        |     A       |     N       |
+| kubelet v1.30    |     X       |     X        |     X       |     N       |
+| scheduler v1.26  |     N       |     X        |     X       |     X       |
+| scheduler v1.27  |     N       |     N        |     X       |     X       |
+| scheduler v1.28  |     X       |     N        |     B       |     X       |
+| scheduler v1.29  |     X       |     X        |     N       |     N       |
+| scheduler v1.30  |     X       |     X        |     X       |     N       |
+| kubectl v1.26    |     C       |     X        |     X       |     X       |
+| kubectl v1.27    |     N       |     N        |     X       |     X       |
+| kubectl v1.28    |     N       |     N        |     N       |     X       |
+| kubectl v1.29    |     X       |     N        |     N       |     N       |
+| kubectl v1.30    |     X       |     X        |     N       |     N       |
+| kubectl v1.31    |     X       |     X        |     X       |     N       |
+
+**X**: Not allowed
+
+**N**: No special handling needed.
+
+**A**: kubelet sets PodStatus.Resize=Infeasible if it sees PodStatus.Resize=Proposed
+       when feature-gate is disabled on kubelet
+
+**B**: Use max(ResourcesAllocated, Requests) if PodStatus.Resize != "" (empty)
```

Comment:
Why scheduler v1.27 + apiserver v1.19 is Not-Allowed?

Reply:
Published K8s version skew policy allows for scheduler version to be upto 1 minor version older.

https://kubernetes.io/releases/version-skew-policy/#kube-controller-manager-kube-scheduler-and-cloud-controller-manager

Reply:
Discussed it at SIG Node weekly earlier. This follows the version skew policy among the components in the control plane. Thanks!